Modules/_zoneinfo.c

```diff
@@ -2243,6 +2246,95 @@ ymd_to_ord(int y, int m, int d)
     return days_before_year + yearday + d;
 }
 
+/* Convert a generic PyObject into a PyDateTime_DateTime object. */
+static PyDateTime_DateTime *
+as_datetime(PyObject *dt)
+{
+    if (PyDateTime_CheckExact(dt)) {
+        Py_INCREF(dt);
+        return (PyDateTime_DateTime *) dt;
+    }
+
+    int year, month, day;
+    int hour, minute, second;
+    unsigned char fold;
+
+    PyObject *num = PyObject_GetAttrString(dt, "year");
+    if (num == NULL) {
+        return NULL;
+    }
+    year = PyLong_AsLong(num);
+    Py_DECREF(num);
+    if (year == -1) {
+        return NULL;
+    }
+
+    num = PyObject_GetAttrString(dt, "month");
+    if (num == NULL) {
+        return NULL;
+    }
+    month = PyLong_AsLong(num);
+    Py_DECREF(num);
+    if (month == -1) {
+        return NULL;
+    }
+
+    num = PyObject_GetAttrString(dt, "day");
+    if (num == NULL) {
+        return NULL;
+    }
+    day = PyLong_AsLong(num);
+    Py_DECREF(num);
+    if (day == -1) {
+        return NULL;
+    }
+
+    num = PyObject_GetAttrString(dt, "hour");
+    if (num == NULL) {
+        return NULL;
+    }
+    hour = PyLong_AsLong(num);
+    Py_DECREF(num);
+    if (hour == -1) {
+        return NULL;
+    }
+
+    num = PyObject_GetAttrString(dt, "minute");
+    if (num == NULL) {
+        return NULL;
+    }
+    minute = PyLong_AsLong(num);
+    Py_DECREF(num);
+    if (minute == -1) {
+        return NULL;
+    }
+
+    num = PyObject_GetAttrString(dt, "second");
+    if (num == NULL) {
+        return NULL;
+    }
+    second = PyLong_AsLong(num);
+    Py_DECREF(num);
+    if (second == -1) {
+        return NULL;
+    }
+
+    num = PyObject_GetAttrString(dt, "fold");
+    if (num == NULL) {
+        return NULL;
+    }
+    int tmp = PyLong_AsLong(num);
+    Py_DECREF(num);
+    if (tmp > 1) {
+        return NULL;
```

Comment:
**issue:** This returns null (indicating an error) without setting an appropriate error string.